lib/sandbox_results_handler.rb

```diff
@@ -24,17 +24,20 @@ def generate_data_for_frontend(test_output, package_type)
 
   def test_results(test_output)
     # Push test results into exercise's error messages
-    return if test_output['testResults'].empty? || test_output['testResults'].first['successful']
+    return if test_output['testResults'].empty? || test_output['status'] == 'PASSED'
     header = 'Virheet testeissä: '
-    messages = test_output['testResults'][0]['message']
+    messages = ''
```

Comment:
```ruby
messages = test_output['testResults'].each_with_object('') do |test, string|
  string += messages += "#{test['message']}<rivinvaihto>"
end
```

Reply:
Good code.

Reply:
don't understand the requested change, it doesn't seem to compile.

Reply:
Should be better now

Reply:
can't modify frozen string

Reply:
Ah, the good ol' magic comment. Maybe try replacing the empty string with `String.new`?